pkg/kubernetes/client.go

```diff
@@ -0,0 +1,164 @@
+package kubernetes
+
+import (
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"errors"
+	"fmt"
+)
+
+var (
+	emptyListOptions = meta_v1.ListOptions{}
+	emptyGetOptions = meta_v1.GetOptions{}
+)
+
+// Client for Kubernetes and Istio APIs
+// It hides the way it queries each API
+type IstioClient struct {
```

Comment:
Why "IstioClient" since it also handles k8s?

Reply:
Istio is not a different backend than kubernetes.
It's a kubernetes extension, so it made sense to me to call it as the "Extended" type as we ask the kubernetes core domain plus the Istio objects.